pio-scripts/cores3_v17_neopixelbus_patch.py

```
@@ -0,0 +1,519 @@
+    if sibling.is_file():
+        try:
+            text = sibling.read_text(encoding="utf-8")
+        except Exception as exc:
+            raise RuntimeError(
+                f"CoreS3 LCD patch could not read {sibling}: {exc}"
+            ) from exc
+
+        if "NeoEspLcdMonoBuffContext" in text and "gdma_connect" in text:
+            return sibling
+
+    libdeps_root = Path(env.subst("$PROJECT_LIBDEPS_DIR")) / env.subst("$PIOENV")
```

Ruff (0.16.3)

[error] Undefined name `env`

(F821)